hazelcast/include/hazelcast/client/ClientConfig.h

```diff
@@ -286,10 +286,18 @@ namespace hazelcast {
             *  set SEVERE to see only serious errors
             *
             *  Default log level is INFO
+            * @param loggerLevel The log level to be set.
             * @return itself ClientConfig
             */
-            ClientConfig &setLogLevel(LogLevel loggerLevel);
+            ClientConfig &setLogLevel(LoggerLevel::Level loggerLevel);
```

Comment:
Is this new method really necessary? 
From the code, it looks like old one works just fine. Why are we deprecating that and introducing a new one?

Reply:
Because when using the old one it is directly at the hazelcast::client namespace, hence you use for example isEnabled(WARNING), this seems like a global variable but I find it better looking usage like this: isEnabled(LoggerLevel::WARNING). It is really a matter of taste.

Reply:
Keeping the ClientConfig api the same but changing the internal ILogger api and removing the deprecated one since it is an internal API.